sentry_sdk/utils.py

```diff
@@ -658,19 +664,56 @@ def single_exception_from_error_tuple(
     tb,  # type: Optional[TracebackType]
     client_options=None,  # type: Optional[Dict[str, Any]]
     mechanism=None,  # type: Optional[Dict[str, Any]]
+    exception_id=0,  # type: int
+    parent_id=0,  # type: int
+    source=None,  # type: Optional[str]
 ):
     # type: (...) -> Dict[str, Any]
-    mechanism = mechanism or {"type": "generic", "handled": True}
+    """
+    Creates a dict that goes into the events `exception.values` list and is ingestible by Sentry.
+
+    See the Exception Interface documentation for more details:
+    https://develop.sentry.dev/sdk/event-payloads/exception/
+    """
+    exception_value = {}  # type: Dict[str, Any]
+    exception_value["mechanism"] = (
+        mechanism.copy() if mechanism else {"type": "generic", "handled": True}
+    )
+    exception_value["mechanism"]["exception_id"] = exception_id
 
     if exc_value is not None:
         errno = get_errno(exc_value)
     else:
         errno = None
 
     if errno is not None:
-        mechanism.setdefault("meta", {}).setdefault("errno", {}).setdefault(
-            "number", errno
-        )
+        exception_value["mechanism"].setdefault("meta", {}).setdefault(
+            "errno", {}
+        ).setdefault("number", errno)
+
+    if source is not None:
+        exception_value["mechanism"]["source"] = source
+
+    is_root_exception = exception_id == 0
+    if not is_root_exception:
+        exception_value["mechanism"]["parent_id"] = parent_id
+        del exception_value["mechanism"]["handled"]
```

Comment:
why the `del`?

Reply:
The RFC states, that the `handled` should always be taken from the root exception and I thought I make it extra explicit. 

See https://github.com/getsentry/rfcs/blob/main/text/0079-exception-groups.md#issue-titles

Reply:
I would just leave this for the server to process, the SDK can just send those over and have the handled in each separate exception of the group, it's extra metadata that doesn't hurt to have in case we change the product in the future. cc @mattjohnsonpint what do you think?

Reply:
I would just leave this for the server to process, the SDK can just send those over and have the handled in each separate exception of the group, it's extra metadata that doesn't hurt to have in case we change the product in the future. cc @mattjohnsonpint what do you think?

Reply:
The overall "handled" of the issue should come from the root exception, but it's ok if each individual exception has its own handled flag.  Indeed, if an exception is caught and a new exception is thrown, then the first exception (the child) was handled - even if the parent exception wasn't.

Reply:
Ok. Will set handled everywhere.